### PR TITLE
[#57] Integrate home screen user detail & top bar

### LINKED ISSUE #57: [Integrate] As a logged in user, I can see my profile detail in the Home screen
## Why
In the home screen, at the title bar, there is an avatar that comes from the API, we need to map it

## Acceptance Criteria
- Bind Date data
- Bind Profile data to the Home screen accordingly.

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #57

## What happened 👀

Integrate task for the home screen: top bar data + user info
 
## Insight 📝

- Normal integration: binding data from controller to View.
- Fix `http` config for specific domain to be able to load avatar on both iOS and Android
 
## Proof Of Work 📹
The app loads correctly without showing any HTTP error:

### CODE REVIEW

File: lib/pages/home/home_controller.dart [hunk 1]
```diff
@@ -1,15 +1,33 @@
 import 'package:get/get.dart';
+import 'package:intl/intl.dart';
 import 'package:survey/managers/user_manager.dart';
 
+const HOME_DISPLAY_DATE_FORMAT = 'EEEE, MMMM dd';
+const AVATAR_PLACE_HOLDER =
```
Comment (markgravity) [inline lib/pages/home/home_controller.dart:6]: Can we save it into our resources? 🤔

Comment (sleepylee) [reply lib/pages/home/home_controller.dart:6]: we can, a similar idea so I just demonstrate it quickly as the placeholder, for now, 😅  I can update if you guys wanted

Comment (sleepylee) [reply lib/pages/home/home_controller.dart]: updated

**Comment by sleepylee** [comment]
@manh-t @markgravity sorry the previous #62 was merged into here 😢  please help me review as well since the Integration is quite small, only bind data at the home screen

File: lib/pages/home/home_controller.dart [hunk 2]
```diff
@@ -1,15 +1,31 @@
 import 'package:get/get.dart';
+import 'package:intl/intl.dart';
 import 'package:survey/managers/user_manager.dart';
 
+const HOME_DISPLAY_DATE_FORMAT = 'EEEE, MMMM dd';
+
 class HomeController extends GetxController {
+  final currentDateText = "".obs;
+  final currentUserAvatar = "".obs;
```
Comment (manh-t) [inline lib/pages/home/home_controller.dart:9]: I think we should not expose the HomeController's mutable value outside. Imo we should create immutable values and listen the change through it 🙏

File: lib/pages/home/home_page.dart [hunk 3]
```diff
@@ -21,32 +20,35 @@ class HomePage extends StatelessWidget {
...
+                          ],
+                        ),
+                        CircleAvatar(
+                          radius: 22,
+                          backgroundColor: Colors.black12,
+                          foregroundImage:
+                              state.currentUserAvatar.value.isNotEmpty
+                                  ? NetworkImage(state.currentUserAvatar.value)
+                                  : AssetImage(
+                                      'assets/images/avatar_placeholder.png'),
```
Comment (manh-t) [inline lib/pages/home/home_page.dart:47]: I believe that we will re-use the avatar placeholder in Drawer. How about creating constant variable for this string? 😄

Comment (manh-t) [reply lib/pages/home/home_page.dart]: @sleepylee only one comment here 🙋

Comment (sleepylee) [reply lib/pages/home/home_page.dart]: updated, @manh-t 🙏

**Comment by sleepylee** [comment]
@markgravity @manh-t PR is ready for review 🙇

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: README.md [hunk 4]
```diff
@@ -15,6 +15,14 @@ A survey conducting platform built with Flutter
 
     - `$ flutter pub run build_runner build --delete-conflicting-outputs`
 
+- Running unit testing:
+
+    - `$ flutter test test/directory/to/test` 
+    
+- To generate Assets/Font/Image:
+    
+    - `$ flutter packages pub run build_runner build --delete-conflicting-outputs`
```
Comment (manh-t) [inline README.md:24]: no biggies, we can keep same command like above 😄 
```suggestion
    - `$ flutter pub run build_runner build --delete-conflicting-outputs`    
```

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Pull request merged**